test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py

```diff
@@ -0,0 +1,257 @@
+         '~10.2.2.140%251?ver=11.6.0',
+
+         u'https://localhost/mgmt/tm/ltm/virtual/'
+         '~TEST_128a63ef33bc4cf891d684fad58e7f2d'
+         '~SAMPLE_LISTENER?ver=11.6.0'])
+
+
+@pytest.fixture
+def configure_icd():
+    class ConfFake(object):
+        '''minimal fake config object to replace oslo with controlled params'''
+        def __init__(self, params):
+            self.__dict__ = params
+            for k, v in self.__dict__.items():
+                if isinstance(v, unicode):
+                    self.__dict__[k] = v.encode('utf-8')
+
+        def __repr__(self):
+            return repr(self.__dict__)
+
+    def _icd(icd_config):
+        mock_rpc_plugin = mock.MagicMock()
+        mock_rpc_plugin.get_port_by_name.return_value =\
+            [{'fixed_ips': [{'ip_address': '10.2.2.134'}]}]
+        icontroldriver = iControlDriver(ConfFake(icd_config),
+                                        registerOpts=False)
+        icontroldriver.plugin_rpc = mock_rpc_plugin
+        return icontroldriver
+    return _icd
+
+
+def test_featureoff_withsegid_lb(setup_neutronless_test, configure_icd, bigip):
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_OFF)
+    icontroldriver._common_service_handler(SEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS | SEG_DEPENDENT_LB_URIS
+
+
+@pytest.mark.skip(reason="This test will not pass until the feature short'\
+    ' circuits prior to network service initialization, in the absence of'\
+    ' a segid.")
+def test_featureoff_nosegid_lb(setup_neutronless_test, configure_icd, bigip):
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_OFF)
+    icontroldriver._common_service_handler(NOSEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS
+
+
+def test_withsegid_lb(setup_neutronless_test, configure_icd, bigip):
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    icontroldriver._common_service_handler(SEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS | SEG_DEPENDENT_LB_URIS
+
+
+def test_nosegid_lb(setup_neutronless_test, configure_icd, bigip):
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    icontroldriver._common_service_handler(NOSEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS
+
+
+@pytest.mark.skip(reason="This test will not pass until triggering the'\
```

Comment:
This used to work, so please file a github issue and remove the 'skip' decorator so the test executes.